### PR TITLE
Fix `startMasters()` from receiving unchecked exceptions

### PR DESCRIPTION
**Pull request by jenoudet**
`FaultTolerantAlluxioMasterProcess` calls `startMasters` when gaining primacy, which can throw a `NotLeaderException` if the master has already lost the leadership, crashing the master. This PR handles this error

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java [hunk 1]
```diff
@@ -191,7 +191,7 @@ public void closeResource() { }
    *
    * @param isLeader if the Master is leader
    */
-  protected void startMasters(boolean isLeader) {
+  protected void startMasters(boolean isLeader) throws IOException {
```
Comment (ggezer) [inline core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java:194]: This method will never throw `IOException` but only `RuntimeException`.

File: core/server/master/src/main/java/alluxio/master/FaultTolerantAlluxioMasterProcess.java [hunk 2]
```diff
@@ -142,7 +143,12 @@ private boolean gainPrimacy() throws Exception {
         return false;
       }
     }
-    startMasters(true);
+    try {
+      startMasters(true);
+    } catch (UnavailableException e) {
+      LOG.warn("Error starting masters: {}", e.toString());
+      return false;
```
Comment (ggezer) [inline core/server/master/src/main/java/alluxio/master/FaultTolerantAlluxioMasterProcess.java:150]: Need to stop already started masters before returning `false`.

**Review by ggezer — CHANGES_REQUESTED** [review]

**Review by ggezer — APPROVED** [review]

**Comment by jenoudet** [comment]
Failed test: 
Java 11 & Java 8:
```
S3ClientRestApiTest.listAllMyBuckets
```

Java 11:
```
FileSystemMasterRestartIntegrationTest.updateAccessTimeAsyncFlushAfterDelete
BlockMasterRegisterStreamIntegrationTest.reregisterWithDelete
BlockMasterIntegrityIntegrationTest.deleteOrphanedBlocks
```
Java 8:
```
ConfigCheckerIntegrationTest.multiNodes
RocksInodeStoreTest
```

**Comment by ggezer** [comment]
alluxio-bot, merge this please

**Comment by Xenorith** [comment]
alluxio-bot, cherry-pick this to branch-2.7 please

**Pull request merged**